cylc/flow/task_events_mgr.py

```diff
@@ -935,8 +936,29 @@ def _get_events_conf(self, itask, key, default=None):
 
     def _process_job_logs_retrieval(self, schd, ctx, id_keys):
         """Process retrieval of task job logs from remote user@host."""
+        # get a host to run retrieval on
         platform = get_platform(ctx.platform_name)
-        host = get_host_from_platform(platform, bad_hosts=self.bad_hosts)
+        try:
+            host = get_host_from_platform(platform, bad_hosts=self.bad_hosts)
+        except NoHostsError:
+            # All of the platforms hosts have been found to be uncontactable.
+            # Reset the bad hosts to allow retrieval retry to take place.
+            self.bad_hosts -= set(platform['hosts'])
```

Comment:
Are you sure we should reset the property instead of using a local variable that could be passed to `get_host_from_platform()` below?

Reply:
Yes, this is intentional.

Resetting the bad_hosts for the platform here will allow other commands to continue.

For future reference the design intention can be found here: https://github.com/cylc/cylc-flow/issues/3827#issuecomment-829104502